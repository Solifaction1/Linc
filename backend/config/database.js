const path = require('path');

const connection = process.env.NODE_ENV !== 'production' ?
  {
    client: 'sqlite',
    connection: {
      filename: path.join(__dirname, '..', '.tmp/data.db'),
    },
    useNullAsDefault: true,
  } : {
    client: "postgres",
    acquireConnectionTimeout: 120000,
    connection: {
      // host: `/cloudsql/${process.env.INSTANCE_CONNECTION_NAME}`, // for a PostgreSQL database
      host: process.env.DATABASE_HOST, // for a PostgreSQL database
      database: process.env.DATABASE_NAME,
      user: process.env.DATABASE_USERNAME,
      password: process.env.DATABASE_PASSWORD,
      ssl: false,

    },
    useNullAsDefault: true,
    debug: false,
  };

module.exports = ({ }) => ({
  connection: connection,
});
