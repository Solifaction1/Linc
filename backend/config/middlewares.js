module.exports = [
  'strapi::errors',
  {
    name: 'strapi::cors',
    config: {
      enabled: true,
      headers: ['*'],
      origin: ['*']
    }
  },
  'strapi::poweredBy',
  'strapi::logger',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            "cdn.jsdelivr.net",
            "strapi.io",
            "s3.amazonaws.com",
            "editor.unlayer.com",
            process.env.CDN_URL,
          ],
          "script-src": ["'self'", "editor.unlayer.com"],
          "frame-src": ["'self'", "editor.unlayer.com"],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            process.env.CDN_URL,
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
];
