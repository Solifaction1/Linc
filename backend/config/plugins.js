module.exports = ({ env }) => ({
  "strapi-plugin-populate-deep": {
    config: {
      defaultDepth: 5, // Default is 5
    },
  },
  "duplicate-button": true,
  email: {
    config: {
      provider: "mailgun",
      providerOptions: {
        key: env("MAILGUN_API_KEY"), // Required
        domain: env("MAILGUN_DOMAIN"), // Required
        url: "https://api.eu.mailgun.net", //Optional. If domain region is Europe use 'https://api.eu.mailgun.net'
      },
      settings: {
        defaultFrom: "<mg@linc.legal>",
        defaultReplyTo: "<mg@linc.legal>",
      },
    },
  },
  upload: {
    config: {
      provider: "aws-s3",
      providerOptions: {
        accessKeyId: env("AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("AWS_ACCESS_SECRET"),
        region: env("AWS_REGION"),
        params: {
          Bucket: env("AWS_BUCKET"),
        },
      },
      actionOptions: {
        upload: {},
        uploadStream: {},
        delete: {},
      },
    },
  },
  "content-versioning": {
    enabled: false,
  },
  slugify: {
    enabled: false,
    config: {
      contentTypes: {
        blog: {
          field: "slug",
          references: "title",
        },
      },
    },
  },
  "preview-button": {
    enabled: true,
    resolve: "./src/plugins/preview-button", // path to plugin folder
  },
  "email-designer": {
    enabled: true,

    // ⬇︎ Add the config property
    config: {
      editor: {
        // optional - if you have a premium unlayer account
        // projectId: [UNLAYER_PROJECT_ID],

        tools: {
          heading: {
            properties: {
              text: {
                value: "This is the new default text!",
              },
            },
          },
        },
        options: {
          features: {
            colorPicker: {
              presets: ["#D9E3F0", "#F47373", "#697689", "#37D67A"],
            },
          },
          fonts: {
            showDefaultFonts: false,
            /*
             * If you want use a custom font you need a premium unlayer account and pass a projectId number :-(
             */
            customFonts: [
              {
                label: "Anton",
                value: "'Anton', sans-serif",
                url: "https://fonts.googleapis.com/css?family=Anton",
              },
              {
                label: "Lato",
                value: "'Lato', Tahoma, Verdana, sans-serif",
                url: "https://fonts.googleapis.com/css?family=Lato",
              },
              // ...
            ],
          },
          mergeTags: [
            {
              name: "Naam",
              value: "{{= USER.Naam }}",
              sample: "Naam",
            },
            {
              name: "Voornaam",
              value: "{{= USER.Voornaam }}",
              sample: "Voornaam",
            },
            {
              name: "Email",
              value: "{{= USER.Email }}",
              sample: "john@doe.com",
            },
            {
              name: "Telefoonnummer",
              value: "{{= USER.Telefoonnummer }}",
              sample: "0123456789",
            },
            {
              name: "select",
              value: "{{= USER.select }}",
              sample: "select",
            },
            {
              name: "textArea",
              value: "{{= USER.textArea }}",
              sample: "textArea",
            },
            {
              name: "Form ID",
              value: "{{= USER.formId }}",
              sample: "0",
            },
          ],
        },
        appearance: {
          theme: "dark",
          panels: {
            tools: {
              dock: "left",
            },
          },
        },
      },
    },
  },
  seo: {
    enabled: true,
  },
  ckeditor: {
    enabled: true,
    config: {
      plugin: {
        // disable data-theme tag setting //
        // setAttribute:false,
        // disable strapi theme, will use default ckeditor theme //
        // strapiTheme:false,
        // styles applied to editor container (global scope) //
        // styles:`
        // :root{
        //   --ck-color-focus-border:red;
        //   --ck-color-text:red;
        // }
        // `
        styles: `
        .ck-editor__main p,
        .ck.ck-list__item .ck-button .ck-button__label {
          line-height: normal;
        }`,
      },
      editor: {
        // editor default config

        // https://ckeditor.com/docs/ckeditor5/latest/features/markdown.html
        // if you need markdown support and output set: removePlugins: [''],
        // default is
        removePlugins: ["Markdown"],

        // https://ckeditor.com/docs/ckeditor5/latest/features/toolbar/toolbar.html
        toolbar: {
          items: [
            "paragraph",
            "heading1",
            "heading2",
            "heading3",
            "|",
            "bold",
            "italic",
            "underline",
            "fontSize",
            "removeFormat",
            "|",
            "bulletedList",
            "numberedList",
            "alignment",
            "|",
            "StrapiMediaLib",
            "link",
            "horizontalLine",
            "|",
            "sourceEditing",
            "fullScreen",
            "undo",
            "redo",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/font.html
        fontSize: {
          options: [
            9,
            11,
            13,
            {
              title: "default",
              model: "18px",
            },
            17,
            19,
            21,
            27,
            35,
          ],
          supportAllValues: false,
        },
        fontFamily: {
          options: [
            "default",
            "Arial, Helvetica Neue, Helvetica, Source Sans Pro, sans-serif",
            "Courier New, Courier, monospace",
            "Georgia, serif",
            "Lucida Sans Unicode, Lucida Grande, sans-serif",
            "Tahoma, Geneva, sans-serif",
            "Times New Roman, Times, serif",
            "Trebuchet MS, Helvetica, sans-serif",
            "Verdana, Geneva, sans-serif",
            "Roboto, Roboto Black, Roboto Medium, Roboto Light, sans-serif",
          ],
          supportAllValues: true,
        },
        fontColor: {
          columns: 5,
          documentColors: 10,
        },
        fontBackgroundColor: {
          columns: 5,
          documentColors: 10,
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
        // default language: 'en',
        // https://ckeditor.com/docs/ckeditor5/latest/features/images/images-overview.html
        image: {
          resizeUnit: "%",
          resizeOptions: [
            {
              name: "resizeImage:original",
              value: null,
              icon: "original",
            },
            {
              name: "resizeImage:25",
              value: "25",
              icon: "small",
            },
            {
              name: "resizeImage:50",
              value: "50",
              icon: "medium",
            },
            {
              name: "resizeImage:75",
              value: "75",
              icon: "large",
            },
          ],
          toolbar: [
            "toggleImageCaption",
            "imageTextAlternative",
            "imageStyle:inline",
            "imageStyle:block",
            "imageStyle:side",
            "linkImage",
            "resizeImage:25",
            "resizeImage:50",
            "resizeImage:75",
            "resizeImage:original",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/table.html
        table: {
          contentToolbar: [
            "tableColumn",
            "tableRow",
            "mergeTableCells",
            "tableCellProperties",
            "tableProperties",
            "toggleTableCaption",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/headings.html
        heading: {
          options: [
            {
              model: "paragraph",
              title: "Paragraph",
              class: "ck-heading_paragraph",
            },
            {
              model: "heading1",
              view: "h1",
              title: "Heading 1",
              class: "ck-heading_heading1",
            },
            {
              model: "heading2",
              view: "h2",
              title: "Heading 2",
              class: "ck-heading_heading2",
            },
            {
              model: "heading3",
              view: "h3",
              title: "Heading 3",
              class: "ck-heading_heading3",
            },
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/general-html-support.html
        htmlSupport: {
          allow: [
            {
              name: "img",
              attributes: {
                sizes: true,
                loading: true,
              },
            },
          ],
        },
      },
    },
  },
});
