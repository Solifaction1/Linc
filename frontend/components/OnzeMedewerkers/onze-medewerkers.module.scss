@import "styles/vars";
@import 'styles/utilities';

.onzeMedewerkers {
    @extend %page-container;
    grid-template-columns: 1fr;
    gap: calc(#{$spacing-base} * 5);
    text-align: center;

    h2 {
        color: $color-beige-dark;
    }

    .employees {
        overflow: scroll;
        overflow-y: auto;

        max-width: 100%;

        scroll-behavior: smooth;
        transition: transform 0.3s ease-in-out;

        .employeeWrapper {
            display: flex;
            gap: calc(#{$spacing-base} * 3);

            text-align: start;
            transition: all 1s;

            .employee {
                img {
                    object-fit: cover;
                }

                .text {
                    display: flex;
                    flex-direction: column;
                    margin-top: calc(#{$spacing-base} * 5);

                    h6 {
                        font-weight: $font-weight-regular;
                        line-height: 2.7rem;
                    }
                    
                    .name {
                        color: $color-beige-dark;
                    }

                    .telephone {
                        margin-bottom: calc(#{$spacing-base} * 2);
                        color: $color-beige-dark;  
                    }

                    .telephone h6, .rechtsdomein {
                        color: $color-brown;
                    }

                    .icon {
                        width: 3.2rem;
                        height: 3.2rem;
                        margin-right: calc(#{$spacing-base} * 2);
                        margin-bottom: calc(#{$spacing-base} * 2);
                        padding: 0.5rem;
                        border: 1px solid $color-beige-dark;
                        border-radius: 50%;

                        fill: $color-black;

                        path {
                            fill: $color-beige-dark;
                        }

                        &:hover {
                            background: $color-beige-dark-opacity-25;
                        }
                    }
                }
            }
        }
    }

    .buttons {
        display: flex;
        gap: calc(#{$spacing-base} * 1);
        justify-content: flex-end;

        button {
            padding: 8px;
            border-radius: 50%;
        }

        svg {
            display: flex;
        }
    }
}