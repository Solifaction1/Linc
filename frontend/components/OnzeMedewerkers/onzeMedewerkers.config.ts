/* eslint-disable max-lines */
import OnzeMedewerkersData from "@models/onzeMedewerkers.model"

const defaultProps: OnzeMedewerkersData = {
	id: 2,
	title: 'title',
	description: 'description',
	employees: {
		data: [
			{
				id: 0,
				attributes: {
					name: 'name',
					locale: 'locale',
					firstName: 'firstName',
					role: 'role',
					rechtsdomeinen: {
						data: [{
							id: 1,
							attributes: {
								title: 'title',
								description: {
									text: '<p>test</p>'
								},
								image: {
									data: null
								},
								slug: 'slug',
								updatedAt: 'new Date()',
								publishedAt: 'new Date()',
								createdAt: 'new Date()',
								component: [],
								locale: 'locale',
								seo: null,
								body: []
							}
						}]
					},
					telephone: 'telephone',
					slug: 'slug',
					updatedAt: 'new Date()',
					publishedAt: 'new Date()',
					createdAt: 'new Date()',
					color: 'color',
					header: {
						color: 'color',
						image: {
							data: {
								id: 0,
								attributes: {
									name: 'name',
									alternativeText: 'alternativeText',
									width: 876,
									height: 1025,
									hash: 'hash',
									ext: 'ext',
									mime: 'mime',
									size: 0,
									url: 'https://via.placeholder.com/876x1025',
								},
							}
						},
						imagePosition: 'left',
						CTA: null,
						bottomText: { cta: null }
					},
					linkedin: 'https://linkedin.com',
					showInNavigation: true
				}
			}
		]
	}
}

export { defaultProps }
