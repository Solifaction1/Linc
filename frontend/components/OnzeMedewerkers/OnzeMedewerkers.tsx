/* eslint-disable max-lines */
import React, { useEffect, useRef, useState } from 'react'
import Image from 'next/image'
import Link from 'next/link'
import { useMediaQuery } from 'react-responsive'

import OnzeMedewerkersData from '@models/onzeMedewerkers.model'
import SwitchIcon from '@templates/SwitchIcon'
import { Icon } from '@templates/switch-icon.config'
import Button from '@components/Button/Button'
import { BREAKPOINTS } from '@constants/breakpoints'
import { defaultProps } from './onzeMedewerkers.config'
import styles from './onze-medewerkers.module.scss'

const OnzeMedewerkers: React.FC<OnzeMedewerkersData> = (
	props: OnzeMedewerkersData
) => {
	const [currentSlide, setCurrentSlide] = useState(0)
	const [interval, setInterval] = useState(0)
	const timelineRef = useRef<HTMLDivElement>(null)
	const isMobile = useMediaQuery({ maxWidth: BREAKPOINTS.md })
	const isTablet = useMediaQuery({ maxWidth: BREAKPOINTS.lg })

	const handleNext = () => {
		setCurrentSlide((prevSlide) => prevSlide + 1)
	}

	const handleBack = () => {
		setCurrentSlide((prevSlide) => prevSlide - 1)
	}

	useEffect(() => {
		if (isMobile) {
			setInterval(1)
		} else if (isTablet) {
			setInterval(3)
		} else {
			setInterval(4)
		}
	}, [isMobile, isTablet])

	useEffect(() => {
		const slideSize = 29.6
		if (timelineRef.current) {
			timelineRef.current.style.transform = `translateX(-${
				currentSlide * (slideSize - 2) * interval
			}rem)`
		}
	}, [currentSlide, interval])

	return (
		<section className={styles.onzeMedewerkers}>
			<h2>{props.title}</h2>
			<p>{props.description}</p>

			{props.employees.data && (
				<div className={styles.employees}>
					<div className={styles.employeeWrapper} ref={timelineRef}>
						{props.employees.data.map((employee) => (
							<div key={employee.id} className={styles.employee}>
								<Link href={`/employee/${employee.attributes.slug}`}>
									<Image
										src={
											employee.attributes.header.image.data?.attributes.url ??
											''
										}
										alt={
											employee.attributes.header.image.data?.attributes
												.alternativeText ?? ''
										}
										height={321}
										width={268}
										priority
										loader={() => {
											return `/api/custom-loader?url=${encodeURIComponent(
												employee.attributes.header.image.data?.attributes.url ??
													''
											)}?q=${100}`
										}}
									/>
								</Link>

								<div className={styles.text}>
									<h6 className={styles.name}>
										{`${`${employee.attributes.firstName}`}${` ${employee.attributes.name}`}`}
									</h6>
									<a
										href={`tel:${employee.attributes.telephone}`}
										className={styles.telephone}
									>
										<h6>{`Tel: ${employee.attributes.telephone}`}</h6>
									</a>

									{employee.attributes.linkedin && (
										<a href={employee.attributes.linkedin}>
											<SwitchIcon
												icon={Icon.LINKEDIN}
												className={styles.icon}
											/>
										</a>
									)}

									{employee.attributes.rechtsdomeinen.data.map(
										(rechtsdomein) => (
											<p key={rechtsdomein.id} className={styles.rechtsdomein}>
												{rechtsdomein.attributes.title}
											</p>
										)
									)}
								</div>
							</div>
						))}
					</div>
				</div>
			)}

			<div className={styles.buttons}>
				<Button
					handleClick={handleBack}
					buttonType="secondary"
					icon="Arrow left"
					className={styles.backButton}
					disabled={currentSlide === 0}
				/>

				<Button
					handleClick={handleNext}
					buttonType="secondary"
					icon="Arrow right"
					className={styles.backButton}
					disabled={
						props.employees.data.length - currentSlide * interval < interval
					}
				/>
			</div>
		</section>
	)
}

OnzeMedewerkers.defaultProps = defaultProps

export default OnzeMedewerkers
