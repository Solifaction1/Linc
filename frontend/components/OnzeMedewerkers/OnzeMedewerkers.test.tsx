import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import OnzeMedewerkers from './OnzeMedewerkers'
import { defaultProps } from './onzeMedewerkers.config'

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}))

describe('OnzeMedewerkers', () => {
	it('renders OnzeMedewerkers', () => {
		const { container, getByText } = render(
			<OnzeMedewerkers {...defaultProps} />
		)

		expect(container).toMatchSnapshot()

		expect(container.querySelector('section')).toBeVisible()
		expect(container.getElementsByClassName('onzeMedewerkers')[0]).toBeVisible()

		expect(getByText('title', { selector: 'h2' })).toBeVisible()
		expect(getByText('description', { selector: 'p' })).toBeVisible()

		expect(container.getElementsByClassName('employees')[0]).toBeVisible()
		expect(container.getElementsByClassName('employeeWrapper')[0]).toBeVisible()
		expect(container.getElementsByClassName('employee')[0]).toBeVisible()
		expect(container.querySelector('img')).toBeVisible()

		expect(container.getElementsByClassName('text')[0]).toBeVisible()

		expect(container.getElementsByClassName('name')[0]).toBeVisible()
		expect(getByText('firstName name', { selector: 'h6' })).toBeVisible()

		expect(container.querySelector('a[href="tel:telephone"]')).toBeVisible()
		expect(getByText('Tel: telephone', { selector: 'h6' })).toBeVisible()

		expect(
			container.querySelector('a[href="https://linkedin.com"]')
		).toBeVisible()
		expect(container.querySelector('svg')).toBeVisible()
		expect(container.querySelector('path')).toBeVisible()

		expect(container.getElementsByClassName('rechtsdomein')[0]).toBeVisible()
		expect(getByText('title', { selector: 'p' })).toBeVisible()
	})
})
