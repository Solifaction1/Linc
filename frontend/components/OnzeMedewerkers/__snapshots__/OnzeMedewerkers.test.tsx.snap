// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnzeMedewerkers renders OnzeMedewerkers 1`] = `
<div>
  <section
    class="onzeMedewerkers"
  >
    <h2>
      title
    </h2>
    <p>
      description
    </p>
    <div
      class="employees"
    >
      <div
        class="employeeWrapper"
        style="transform: translateX(-0rem);"
      >
        <div
          class="employee"
        >
          <a
            href="/employee/slug"
          >
            <img
              alt="alternativeText"
              data-nimg="1"
              decoding="async"
              height="321"
              src="/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F876x1025?q=100"
              srcset="/api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F876x1025?q=100 1x, /api/custom-loader?url=https%3A%2F%2Fvia.placeholder.com%2F876x1025?q=100 2x"
              style="color: transparent;"
              width="268"
            />
          </a>
          <div
            class="text"
          >
            <h6
              class="name"
            >
              firstName name
            </h6>
            <a
              class="telephone"
              href="tel:telephone"
            >
              <h6>
                Tel: telephone
              </h6>
            </a>
            <a
              href="https://linkedin.com"
            >
              <svg
                class="icon"
                fill="currentColor"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20.47,2H3.53A1.45,1.45,0,0,0,2.06,3.43V20.57A1.45,1.45,0,0,0,3.53,22H20.47a1.45,1.45,0,0,0,1.47-1.43V3.43A1.45,1.45,0,0,0,20.47,2ZM8.09,18.74h-3v-9h3ZM6.59,8.48h0a1.56,1.56,0,1,1,0-3.12,1.57,1.57,0,1,1,0,3.12ZM18.91,18.74h-3V13.91c0-1.21-.43-2-1.52-2A1.65,1.65,0,0,0,12.85,13a2,2,0,0,0-.1.73v5h-3s0-8.18,0-9h3V11A3,3,0,0,1,15.46,9.5c2,0,3.45,1.29,3.45,4.06Z"
                />
              </svg>
            </a>
            <p
              class="rechtsdomein"
            >
              title
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        class="button false secondary md noLabel backButton beige"
        disabled=""
        type="button"
      >
        <p
          class="label"
        />
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17,11H9.41l3.3-3.29a1,1,0,1,0-1.42-1.42l-5,5a1,1,0,0,0-.21.33,1,1,0,0,0,0,.76,1,1,0,0,0,.21.33l5,5a1,1,0,0,0,1.42,0,1,1,0,0,0,0-1.42L9.41,13H17a1,1,0,0,0,0-2Z"
          />
        </svg>
      </button>
      <button
        class="button false secondary md noLabel backButton beige"
        disabled=""
        type="button"
      >
        <p
          class="label"
        />
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.92,11.62a1,1,0,0,0-.21-.33l-5-5a1,1,0,0,0-1.42,1.42L14.59,11H7a1,1,0,0,0,0,2h7.59l-3.3,3.29a1,1,0,0,0,0,1.42,1,1,0,0,0,1.42,0l5-5a1,1,0,0,0,.21-.33A1,1,0,0,0,17.92,11.62Z"
          />
        </svg>
      </button>
    </div>
  </section>
</div>
`;
